agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1511,3 +1515,50 @@ func (dg *dockerGoClient) loadImage(ctx context.Context, reader io.Reader) error
 	}
 	return err
 }
+
+func (dg *dockerGoClient) CreateContainerExec(ctx context.Context, containerID string, execConfig types.ExecConfig, timeout time.Duration) (*types.IDResponse, error) {
```

Comment:
I want to be more opinionated about what we have in the function signature here.  the `types.ExecConfig` being required always is a burden.  Like we know we don't care about the stdio streams/tty or interactive mode and we know we'll always be detached.  I think you need to decide which bits of the ExecConfig are necessary here (user?  command array? and update the signature accordingly.  Also see the comments here about how to use variadic arguments https://github.com/aws/amazon-ecs-agent/pull/2622

Reply:
I think `types.ExecConfig` arg will always be required here (to create exec config, min required field would be command?) 
I'm ok with either adding only the required fields for now and renaming the method accordingly or leave this as is and support other fields for the future as well.

Reply:
I think using execConfig as is in arguments, gives us more control. We have done that for dockerConfig and dockerHostConfig in ContainerCreate dockerapi (https://github.com/aws/amazon-ecs-agent/blob/dev/agent/dockerclient/dockerapi/docker_client.go#L506). Moreover, I dont think we would know what User and Privileged field be unless we inherit it from dockerConfig and dockerHostConfig for each container.
We will have more control over sending execConfig for different useCases if we pass execConfig as a whole in arguments.
Please let me know if you agree.

Reply:
I agree given that ContainerCreate uses the same pattern.